src/uct/sm/cma/cma_ep.c

```diff
@@ -41,23 +41,27 @@ UCS_CLASS_DEFINE_DELETE_FUNC(uct_cma_ep_t, uct_ep_t);
                     (_rkey))
 
 ucs_status_t uct_cma_ep_put_zcopy(uct_ep_h tl_ep, const void *buffer, size_t length,
-                                   uct_mem_h memh, uint64_t remote_addr,
-                                   uct_rkey_t rkey, uct_completion_t *comp)
+                                  uct_mem_h memh, uint64_t remote_addr,
+                                  uct_rkey_t rkey, uct_completion_t *comp)
 {
-    ssize_t delivered;
+    ssize_t ret;
+    ssize_t delivered = 0;
+    struct iovec local_iov;
+    struct iovec remote_iov;
     uct_cma_ep_t *ep = ucs_derived_of(tl_ep, uct_cma_ep_t);
-    struct iovec local_iov  = {.iov_base = (void *)buffer,
-                               .iov_len = length};
-    struct iovec remote_iov = {.iov_base = (void *)remote_addr,
-                               .iov_len = length};
 
-    delivered = process_vm_writev(ep->remote_pid, &local_iov, 1, &remote_iov, 1, 0);
-    if (ucs_unlikely(delivered != length)) {
-        ucs_error("process_vm_writev delivered %zu instead of %zu, error message %s",
-                  delivered < 0 ? 0 : delivered, length,
-		  delivered < 0 ? strerror(errno): "Unexpected CMA behaviour");
-        return UCS_ERR_IO_ERROR;
-    }         
+    do {
```

Comment:
can we unite this to one function which receives a function pointer to either process_vm_read or process_vm_write?